gmso/formats/lammpsdata.py

```diff
@@ -183,48 +183,159 @@ def write_lammpsdata(topology, filename, atom_style="full"):
+                                .value
+                                / scaling_factor,
+                                bond_type.parameters["r_eq"]
+                                .in_units(u.Unit("angstrom"))
+                                .value,
+                            )
+                        )
+                    else:
+                        raise ValueError(
+                            'Bond Style "{}" is invalid or is not currently supported'.format(
+                                bond_type.expression
+                            )
+                        )
+
+            if topology.angles:
+                data.write("\nAngle Coeffs\n\n")
+                for idx, angle_type in enumerate(topology.angle_types):
+                    # expected lammps harmonic angle expression
+                    angle_expression = "k * (theta - theta_eq)**2"
+
+                    if sympify(angle_type.expression).equals(
+                        sympify(angle_expression)
+                    ):
+                        data.write(
+                            "{}\t{:.5f}\t{:.5f}\n".format(
+                                idx + 1,
+                                angle_type.parameters["k"]
+                                .in_units(u.Unit("kcal/mol/radian**2"))
+                                .value,
+                                angle_type.parameters["theta_eq"]
+                                .in_units(u.Unit("degree"))
+                                .value,
+                            )
+                        )
+                    elif sympify(angle_type.expression).equals(
+                        sympify("0.5 * k * (theta - theta_eq)**2")
+                    ):
+                        data.write(
+                            "{}\t{:.5f}\t{:.5f}\n".format(
+                                idx + 1,
+                                angle_type.parameters["k"]
+                                .in_units(u.Unit("kcal/mol/radian**2"))
+                                .value
+                                / 2.0,
+                                angle_type.parameters["theta_eq"]
+                                .in_units(u.Unit("degree"))
+                                .value,
+                            )
+                        )
+                    elif (
+                        scaling_factor := simplify(angle_expression)
+                        / simplify(angle_type.expression)
+                    ).is_real:
+                        data.write(
+                            "{}\t{:.5f}\t{:.5f}\n".format(
+                                idx + 1,
+                                angle_type.parameters["k"]
+                                .in_units(u.Unit("kcal/mol/radian**2"))
+                                .value
+                                / scaling_factor,
+                                angle_type.parameters["theta_eq"]
+                                .in_units(u.Unit("degree"))
```

Comment:
Need to double check the unit. Per the lammps docs (https://docs.lammps.org/angle_harmonic.html), it mentioned the angles are converted to radians internally, so i am not sure if K also need to be specified in kcal/mol/degree^2 (and the conversion will happen in lammps).

Reply:
Theta is degrees and is converted to radians after being read in by lamps; k is not converted in the code and needs to be per radian squared.

https://github.com/CFDEMproject/LAMMPS/blob/a92e8c271bcb7add8245f5a32f2920fbe35bacc9/src/MOLECULE/angle_harmonic.cpp#L178-L184

Reply:
Good to know!